Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -382,20 +396,6 @@ open class NavigationMapView: UIView {
         navigationCamera.follow()
     }
     
-    func updateUserCourseViewWithAccuracy() {
-        let frame = CGRect(origin: .zero, size: 75.0)
-        let isHidden = userCourseView.isHidden
-        switch userLocationStyle {
-        case .courseView(let courseView):
-            userCourseView = reducedAccuracyActivatedMode ? UserHaloCourseView(frame: frame) : courseView
-        case .puck2D(_):
-            userCourseView = reducedAccuracyActivatedMode ? UserHaloCourseView(frame: frame) : UserPuckCourseView(frame: frame)
-        case .puck3D(_):
-            userCourseView = reducedAccuracyActivatedMode ? UserHaloCourseView(frame: frame) : UserPuckCourseView(frame: frame)
-        }
```

Comment:
I think in current implementation we need to tackle case when user switches `reducedAccuracyActivatedMode` to `true` and then back to `false` while using either 2D or 3D puck. In current implementation we'll fallback to `UserPuckCourseView`, which is not correct.